internal/controller/datadogagent/controller_reconcile_v2_common_test.go

```diff
@@ -215,9 +226,589 @@ func Test_shouldCheckCreateStrategyStatus(t *testing.T) {
+				DatadogAgentProfileEnabled: false,
+			},
+			profile:              &testProfile,
+			ddaLastUpdateTime:    now,
+			now:                  now,
+			existingObjects:      nil,
+			expectedShouldUpdate: true,
+			errorMessage:         nil,
+		},
+		{
+			name: "Profiles nil",
+			reconcilerOptions: ReconcilerOptions{
+				ExtendedDaemonsetOptions: agent.ExtendedDaemonsetOptions{
+					Enabled: true,
+				},
+				DatadogAgentProfileEnabled: true,
+			},
+			profile:              nil,
+			ddaLastUpdateTime:    now,
+			now:                  now,
+			existingObjects:      nil,
+			expectedShouldUpdate: true,
+			errorMessage:         nil,
+		},
+		{
+			name: "Default profile",
+			reconcilerOptions: ReconcilerOptions{
+				ExtendedDaemonsetOptions: agent.ExtendedDaemonsetOptions{
+					Enabled: true,
+				},
+				DatadogAgentProfileEnabled: true,
+			},
+			profile: &v1alpha1.DatadogAgentProfile{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "default",
+					Namespace: "",
+				},
+			},
+			ddaLastUpdateTime:    now,
+			now:                  now,
+			existingObjects:      nil,
+			expectedShouldUpdate: true,
+			errorMessage:         nil,
+		},
+		{
+			name: "No EDSs exist",
+			reconcilerOptions: ReconcilerOptions{
+				ExtendedDaemonsetOptions: agent.ExtendedDaemonsetOptions{
+					Enabled: true,
+				},
+				DatadogAgentProfileEnabled: true,
+			},
+			profile:              &testProfile,
+			ddaLastUpdateTime:    now,
+			now:                  now,
+			existingObjects:      []client.Object{},
+			expectedShouldUpdate: false,
+			errorMessage:         nil,
+		},
+		{
+			name: "Canary paused",
+			reconcilerOptions: ReconcilerOptions{
+				ExtendedDaemonsetOptions: agent.ExtendedDaemonsetOptions{
+					Enabled: true,
+				},
+				DatadogAgentProfileEnabled: true,
+			},
+			profile:           &testProfile,
+			ddaLastUpdateTime: now,
+			now:               now,
+			existingObjects: []client.Object{
+				&edsdatadoghqv1alpha1.ExtendedDaemonSet{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "test-eds",
+						Namespace: testNS,
+						Annotations: map[string]string{
+							edsdatadoghqv1alpha1.ExtendedDaemonSetCanaryPausedAnnotationKey: "true",
+						},
+						Labels: testEDSLabels,
+					},
+				},
+			},
+			expectedShouldUpdate: false,
+			errorMessage:         nil,
+		},
+		{
+			name: "Active canary",
+			reconcilerOptions: ReconcilerOptions{
+				ExtendedDaemonsetOptions: agent.ExtendedDaemonsetOptions{
+					Enabled: true,
+				},
+				DatadogAgentProfileEnabled: true,
+			},
+			profile:           &testProfile,
+			ddaLastUpdateTime: now,
+			now:               now,
+			existingObjects: []client.Object{
+				&edsdatadoghqv1alpha1.ExtendedDaemonSet{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "test-eds",
+						Namespace: testNS,
+						Labels:    testEDSLabels,
+					},
+					Status: edsdatadoghqv1alpha1.ExtendedDaemonSetStatus{
+						Canary: &edsdatadoghqv1alpha1.ExtendedDaemonSetStatusCanary{},
+					},
```

Comment:
```suggestion
					Status: edsdatadoghqv1alpha1.ExtendedDaemonSetStatus{
						Canary: &edsdatadoghqv1alpha1.ExtendedDaemonSetStatusCanary{
							ReplicaSet: "test-ers-2",
							Nodes:      []string{"node-foo"},
						},
					},
```
nit: Not entirely sure about the indent with Github UI, but suggestion is to have a "real" canary status object, not just for the sake of the test passing since it's not nil.